ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/DiskPolicy.hs

```diff
@@ -61,7 +62,11 @@ data DiskPolicy = DiskPolicy {
       --   blocks had to be replayed.
       --
       -- See also 'defaultDiskPolicy'
-    , onDiskShouldTakeSnapshot :: Maybe DiffTime -> Word64 -> Bool
+    , onDiskShouldTakeSnapshot ::
+           Maybe DiffTime
+        -> Word64
+        -> WithOrigin SlotNo
```

Comment:
The `Origin` case should always be a `False` decision; I don't anticipate ever wanting a snapshot for the genesis/initial ledger state. (This is why `OnDisk.takeSnapshot` is a no op in the `Origin` case.)

So my instinct is that this argument should be just `SlotNo`.